src/client.rs

```diff
@@ -1,49 +1,61 @@
 use crate::auth_layer::{AuthClientLayer, AuthClientService};
-use crate::metrics::ClientMetrics;
 use crate::server::{EngineApiClient, PayloadSource};
 use alloy_primitives::B256;
 use alloy_rpc_types_engine::{
     ExecutionPayload, ExecutionPayloadV3, ForkchoiceState, ForkchoiceUpdated, JwtError, JwtSecret,
     PayloadId, PayloadStatus,
 };
 use clap::{Parser, arg};
-use http::{StatusCode, Uri};
-use jsonrpsee::core::{ClientError, RpcResult};
+use http::Uri;
 use jsonrpsee::http_client::transport::HttpBackend;
 use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
-use jsonrpsee::types::ErrorCode;
+use jsonrpsee::types::ErrorObjectOwned;
 use op_alloy_rpc_types_engine::{OpExecutionPayloadEnvelopeV3, OpPayloadAttributes};
+use opentelemetry::trace::SpanKind;
 use paste::paste;
 use std::path::PathBuf;
-use std::sync::Arc;
-use std::time::{Duration, Instant};
+use std::time::Duration;
 use thiserror::Error;
-use tracing::error;
+use tracing::{error, info, instrument};
+
+pub(crate) type ClientResult<T> = Result<T, ClientError>;
 
 #[derive(Error, Debug)]
-pub enum ExecutionClientError {
+pub(crate) enum ClientError {
     #[error(transparent)]
-    HttpClient(#[from] jsonrpsee::core::client::Error),
+    Jsonrpsee(#[from] jsonrpsee::core::client::Error),
+    #[error("Invalid payload: {0}")]
+    InvalidPayload(String),
     #[error(transparent)]
     Io(#[from] std::io::Error),
     #[error(transparent)]
     Jwt(#[from] JwtError),
 }
 
+impl From<ClientError> for ErrorObjectOwned {
+    fn from(err: ClientError) -> Self {
+        match err {
+            ClientError::Jsonrpsee(jsonrpsee::core::ClientError::Call(error_object)) => {
+                error_object
+            }
+            // Status code 13 == internal error
+            e => ErrorObjectOwned::owned(13, e.to_string(), Option::<()>::None),
```

Comment:
is there a way to map different codes for different ClientErrors? also can probably extract the code into a const

Reply:
Yeah certainly we can do that. The above impl is similar to how it was done before, so perhaps that improvement can be left for another PR?

Let's create an issue for it and spec out some error codes.

Reply:
#147